core/pkg/artifacts/linker.go

```diff
@@ -24,8 +25,17 @@ func (al *ArtifactLinker) Link() error {
 	portfolioName := al.LinkArtifact.PortfolioName
 	portfolioEntity := al.LinkArtifact.PortfolioEntity
 	portfolioProject := al.LinkArtifact.PortfolioProject
+	organization := al.LinkArtifact.PortfolioOrganization
 	var portfolioAliases []gql.ArtifactAliasInput
 
+	var err error
+	if IsArtifactRegistryProject(portfolioProject) {
+		portfolioEntity, err = al.resolveOrgEntityName(portfolioEntity, organization)
```

Comment:
Why does this overwrite `portfolioEntity`, not `organization`? I thought `resolveOrgEntityName` returns the org name?

Reply:
Sorry i know this is pretty confusing here is the run down.

We have our traditional orgs. This is in the org table and what you see in the org selector for example.
<img width="359" alt="Screenshot 2024-10-07 at 12 58 05" src="https://github.com/user-attachments/assets/731baed5-627b-4b20-aae6-8e3be3dc1560">

A few months ago we(mainly platform identity team) made the initiative to have every org in the org table to also have an entry in the `Entity` table. This will allow projects to now be under an org. 
In the past the `Entity` table only had teams, and personal entities. Now after the migration for every org in the org table we also have a `org entity`. This is just an org entity that points the org in the org table, and it has the type `organization` instead of `team`. The problem with this migration was that if you have an org called `wandb` and you wanted to create an org entity entry of that org, but you look in the Entities table and already see a team entity called `wandb`, the names have to be unique so we had to slug the new org entity name which resulted in having a name like `wandb_K03HIW` for example. This is really hard for the user to memorize and find so we are trying to have a shorthand method so the user doesn't have to put in an org entity, or they can put in the display org, or the org name that they see in the screenshot above since thats probably what most users are expecting their org name to be.

To summarize we are still fetching/linking artifacts via the entity, project, collection name etc, we just are using the org-entity instead for Registry artifacts (since Registry artifacts are under the org-entity). Its can sometimes be hard for the user to know what their org-entity is since there is a chance it got slugified so we are also accepting the display org (in the screenshot) or the user shorthanding and not putting any org at all and fetching the run/artifact's team entity and fetching the org entity. Where this code comes in is we take that info the user gives us and spits out the org entity